parser.y

```diff
@@ -1468,6 +1513,21 @@ AlterTableSpec:
 		parser.lastErrorAsWarn()
 	}
 
+AllOrPartitionNameList:
+	"ALL"
+	{
+		$$ = &ast.AllOrPartitionNames{
+			All: true,
```

Comment:
```suggestion
			$$ = nil,
```
if we meet `ALL`, we can return nil. so we can remove `ast.AllOrPartitionNames`

Reply:
or we could use `[2]interface{}` to hold two kinds of value.

Reply:
Thx.